Resources/Prototypes/Entities/Structures/Power/Generation/Tesla/energyball.yml

```
@@ -0,0 +1,115 @@
+- type: entity
+  id: BaseEnergyBall
+  abstract: true
+  components:
+  - type: Clickable
+  - type: Physics
+    bodyType: Dynamic
+    bodyStatus: InAir
+  - type: CanMoveInAir
+  - type: ChasingWalk
+    minSpeed: 1
+    maxSpeed: 3
+    chasingComponent: LightningTarget
+  - type: AmbientSound
+    volume: 3
+    range: 15
+    sound:
+      path: /Audio/Effects/tesla.ogg
+  - type: PointLight
+    enabled: true
+    radius: 5
+    color: "#4080FF"
+  - type: Appearance
+  - type: LightningArcShooter
+    arcDepth: 2
+    maxLightningArc: 1
+    shootMinInterval: 4
+    shootMaxInterval: 10
+    shootRange: 3
+    lightningPrototype: ChargedLightning
+
+- type: entity
+  id: TeslaEnergyBall
+  parent: BaseEnergyBall
+  name: ball lightning
+  description: A giant ball of pure energy. The space around it is humming and melting.
+  components: 
+  - type: Fixtures
+    fixtures:
+      TeslaCollider:
+        shape:
+          !type:PhysShapeCircle
+            radius: 0.65
+        hard: true
+        restitution: 0.8
+        density: 99999 # to do: remove the interaction with the explosions. Now the explosions can push the Tesla very far away from the station.
+        mask:
+        - Opaque
+        layer:
+        - GlassLayer
+  - type: TeslaEnergyBall
+    energyLoss: 3
+    spawnProto: TeslaMiniEnergyBall 
+    soundCollapse:
+      path: /Audio/Effects/tesla_collapse.ogg
+      params:
+        variation: 0.3
+  - type: LightningArcShooter
+    arcDepth: 3
+    maxLightningArc: 4
+    shootMinInterval: 2
+    shootMaxInterval: 5
+    shootRange: 7
+    lightningPrototype: HyperchargedLightning
```

```suggestion
    lightningPrototype: Lightning
```
This is why the lightning is having some issues with sprites.

---

Well, I just needed some red dangerous high damage lightning bolts. Well, the fact that the beam system draws them incorrectly, I think, is not a problem of tesla, but a problem of the beam system. It should be possible to write any type of lightning here, and it MUST work, otherwise why this field is needed at all?

---

I will replace now with regular Lightning with a note to return after the beams are fixed.